Tests/UnitTests/Networking/HTTPClientTests.swift

```diff
@@ -1664,6 +1665,145 @@ final class HTTPClientTests: BaseHTTPClientTests<MockETagManager> {
+            httpVersion: nil,
+            headerFields: [HTTPClient.ResponseHeader.eTag.rawValue: ""]
+        )!
+
+        backoffPeriod = self.client.calculateRetryBackoffTime(forResponse: httpURLResponse, retryCount: 1)
+        expect(backoffPeriod).to(equal(TimeInterval(0)))
+    }
+
+    func testUsesBackoffIfFirstRetryAndETagPresent() {
+        var httpURLResponse = HTTPURLResponse(
+            url: URL(string: "api.revenuecat.com")!,
+            statusCode: HTTPStatusCode.tooManyRequests.rawValue,
+            httpVersion: nil,
+            headerFields: [:]
+        )!
+
+        var backoffPeriod = self.client.calculateRetryBackoffTime(forResponse: httpURLResponse, retryCount: 1)
+        expect(backoffPeriod).to(equal(TimeInterval(0)))
+
+        httpURLResponse = HTTPURLResponse(
+            url: URL(string: "api.revenuecat.com")!,
+            statusCode: HTTPStatusCode.tooManyRequests.rawValue,
+            httpVersion: nil,
+            headerFields: [HTTPClient.ResponseHeader.eTag.rawValue: "some-etag-value"]
+        )!
+
+        backoffPeriod = self.client.calculateRetryBackoffTime(forResponse: httpURLResponse, retryCount: 1)
+        expect(backoffPeriod).to(equal(TimeInterval(0)))
+    }
+
+    func testUsesServerProvidedBackoffIfPresent() {
+        let retryAfterMilliseconds = 100
+
+        let httpURLResponse = HTTPURLResponse(
+            url: URL(string: "api.revenuecat.com")!,
+            statusCode: HTTPStatusCode.tooManyRequests.rawValue,
+            httpVersion: nil,
+            headerFields: [
+                HTTPClient.ResponseHeader.retryAfter.rawValue: "\(retryAfterMilliseconds)"
+            ]
+        )!
+
+        let backoffPeriod = self.client.calculateRetryBackoffTime(forResponse: httpURLResponse, retryCount: 2)
+        expect(backoffPeriod).to(equal(TimeInterval(0.1)))  // 0.1s == 100ms
+    }
+
+    func testUsesDefaultBackoffIfServerProvidedBackoffMissing() {
+        let httpURLResponse = HTTPURLResponse(
+            url: URL(string: "api.revenuecat.com")!,
+            statusCode: HTTPStatusCode.tooManyRequests.rawValue,
+            httpVersion: nil,
+            headerFields: [:]
+        )!
+
+        let backoffPeriod = self.client.calculateRetryBackoffTime(forResponse: httpURLResponse, retryCount: 2)
+        expect(backoffPeriod).to(equal(TimeInterval(0.5)))  // 0.1s == 100ms
+    }
+
+    func testDefaultRetryBackoffPeriods() {
+        expect(
+            self.client.calculateDefaultExponentialBackoffTimeInterval(withRetryCount: 0)
+        ).to(equal(0))
+
+        expect(
+            self.client.calculateDefaultExponentialBackoffTimeInterval(withRetryCount: 1)
+        ).to(equal(0.25))
+
+        expect(
+            self.client.calculateDefaultExponentialBackoffTimeInterval(withRetryCount: 2)
+        ).to(equal(0.5))
+
+        expect(
+            self.client.calculateDefaultExponentialBackoffTimeInterval(withRetryCount: 3)
+        ).to(equal(1))
+    }
+
+}
```

Comment:
You can also use OHTTPStubs to fake responses from the backend and test. 

The files named `somethingBackendTests` have plenty of examples you can mostly reuse, especially `BackendPostReceiptDataTests`

Reply:
feels like we should get at least one backend post receipt test in with the exact case we want to cover (first request is a 429, second goes through).

We can also use our BackendIntegrationTests suite to actually force a 429, remove the forced 429, post something to the backend, ensure it went through. 

`OfflineStoreKitIntegrationTests` has some examples of doing something similar